File: src/arduinobot_cpp_examples/src/simple_subscriber.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>

using std::placeholders::_1;

class SimpleSubscriber : public rclcpp::Node
{

public:
SimpleSubscriber(): Node("simple_subscriber")
{
sub_ = create_subscription<std_msgs::msg::String>("Chatter", 10, std::bind(& SimpleSubscriber::msgCallbask, this, _1));

}
void msgCallbask(const std_msgs::msg::String &msg) const
{
RCLCPP_INFO_STREAM(get_logger(), "I Heard: "<<msg.data.c_str());
}

private:
rclcpp::Subscription<std_msgs::msg::String>::SharedPtr sub_;
};

int main(int argc, char*argv[])
{
                        rclcpp::init(argc, argv);
                        auto node = std::make_shared<SimpleSubscriber>();
                        rclcpp::spin(node);
                        rclcpp::shutdown();
                        return 0;
}

File: src/arduinobot_py_examples/arduinobot_py_examples/simple_subscriber.py
import rclpy
from rclpy.node import Node 
from std_msgs.msg import String
from std_msgs.msg import String

class SimpleSubscriber(Node):
        def __init__(self):
                super().__init__("simple_subscriber")
                self.sub_ = self.create_subscription (String, "Chatter", self.msgCallback, 10)

        def msgCallback(self, msg):
                self.get_logger().info("I heard : %s " % msg.data)



def main():
        rclpy.init()
        simple_subscriber = SimpleSubscriber()
        rclpy.spin(simple_subscriber)  ## Keep the node active and the time and node running
        simple_subscriber.destroy_node()
        rclpy.shutdown()


if __name__ == '__main__':
        main()
